### DISCUSSION TITLE
Timing problems?

### DISCUSSION THREAD

**Discussion by vjbutt** [JAndrassy/WiFiEspAT#75] [Q&A]
This library looks like a very good idea, but unfortunately I have problems getting it to work properly.  

The setup that I am testing with is an Arduino Uno board + an ESP8266 ESP-01 module with a 1MByte flash.  The ESP-01 is loaded with the original AT version 1.7.5.0 firmware.

I am using the SoftwareSerial library to connect to the ESP-01 with a baud rate of 9600.  However, using the library as is, I get problems getting the right response from the ESP-01.  For example, even the supplied example program CheckFirmware doesn't work properly.  I managed to get this example working by introducing a delay of 30ms after the AT+RST command and in the functions EspAtDrvClass::sendCommand and EspAtDrvClass::simpleCommand before the serial buffer replies are read.

Alas, the above changes were not enough to make a more complicated example work, such as for example WebServer.  The only way how I managed to get this working, sort of, is by downloading a previous firmware (AT version 1.7.4.0) and only if I enable LOG_LEVEL_DEBUG.  However, even if I go to one level of debugging below this, i.e. LOG_LEVEL_INFO, the WebServer example doesn't work properly (basically the IP of the server is reported incorrectly - 0.0.0.0 and/or the client is not able to connect to the webserver, even when the server IP is reported correctly).  So there must be some timing issues.  But now it is a bit more difficult for me to debug, since I can't enable LOG_LEVEL_DEBUG (because then the program works properly!).  

Any hints of what I can do to get this library working with my setup?

**Comment by JAndrassy** [comment]
did you set the baud rate at the AT firmware side?

**Reply by vjbutt** [reply]
Yes, I did.  On the ESP-01 this was set to 9600, the same setting as used with the SoftwareSerial library.
